readthedocs/embed/tests/test_api.py

```diff
@@ -52,8 +57,49 @@ def _get_html_content(self, html_file):
         section_content = [PyQuery(html_file.open().read()).outerHtml()]
         return section_content
 
+    def test_invalid_arguments(self, client):
+        urls = [
+            f'?project={self.project.slug}&version={self.version.slug}',
+            f'?project={self.project.slug}&version={self.version.slug}&section=foo',
+        ]
+
+        api_endpoint = reverse('api_embed')
+        for url in urls:
+            r = client.get(api_endpoint + url)
+            assert r.status_code == status.HTTP_400_BAD_REQUEST
+
     @mock.patch('readthedocs.embed.views.build_media_storage')
-    def test_embed_unknown_section(self, storage_mock):
+    def test_valid_arguments(self, storage_mock, client):
+        json_file = data_path / 'sphinx/latest/page.json'
+        html_file = data_path / 'sphinx/latest/page.html'
+
+        self._patch_sphinx_json_file(
+            storage_mock=storage_mock,
+            json_file=json_file,
+            html_file=html_file,
+        )
+
+        urls = (
+            # URL only
+            'url=https://project.readthedocs.io/en/latest/index.html#title-one',
+            'url=http://project.readthedocs.io/en/latest/index.html#title-one',
+            'url=http://project.readthedocs.io/en/latest/#title-one',
+            'url=http://project.readthedocs.io/en/latest/index.html?foo=bar#title-one',
+            'url=http://project.readthedocs.io/en/latest/?foo=bar#title-one',
+
+            # doc & path
+            f'project={self.project.slug}&version={self.version.slug}&path=index.html&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&path=/index.html&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&doc=index&section=title-one',
+            f'project={self.project.slug}&version={self.version.slug}&path=index.html&doc=index&section=title-one',
+        )
```

Comment:
Might be nicer to construct these from kwargs and url encode them, so it's easier to tell what we're changing between URL's. Maybe that's too much work tho.

Reply:
turns our the test client accepts a dic for the query strings of get